PowerShellTools/ServiceManagement/ConnectionManager.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Diagnostics;
+using PowerShellTools.Common;
+using PowerShellTools.Common.ServiceManagement.IntelliSenseContract;
+
+namespace PowerShellTools.ServiceManagement
+{
+    /// <summary>
+    /// Manage the process and channel creation.
+    /// </summary>
+    internal sealed class ConnectionManager
+    {
+        private IPowershellIntelliSenseService _powershellServiceChannel;
+        private int _hostProcessId;
+
+        public ConnectionManager()
+        {            
+            OpenClientConnection();
+        }
+
+        /// <summary>
+        /// The service channel we need.
+        /// </summary>
+        public IPowershellIntelliSenseService PowershellServiceChannel

```

Comment:
Done.

From: Eric Deng [mailto:notifications@github.com]
Sent: Wednesday, January 21, 2015 11:19 AM
To: Microsoft/poshtools
Cc: Huanhuan Sun
Subject: Re: [poshtools] Infrastructure for out-of-proc, includes initializing a process and creating a WCF channel for servicing (#36)


In PowerShellTools/ServiceManagement/ConnectionManager.cs<https://github.com/Microsoft/poshtools/pull/36#discussion_r23325676>:

> +    /// Manage the process and channel creation.

> +    /// </summary>

> +    internal sealed class ConnectionManager

> +    {

> +        private IPowershellIntelliSenseService _powershellServiceChannel;

> +        private int _hostProcessId;

> +

> +        public ConnectionManager()

> +        {

> +            OpenClientConnection();

> +        }

> +

> +        /// <summary>

> +        /// The service channel we need.

> +        /// </summary>

> +        public IPowershellIntelliSenseService PowershellServiceChannel

You might want to rename "PowershellServiceChannel" into "PowershellIntelliSenseServiceChannel", coz that is going to be the naming convention debugging channel would follow.

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/poshtools/pull/36/files#r23325676>.